### PR TITLE
Subject Alt Name Matching

### PR DESCRIPTION
**Pull request by ejohnstown**
1. Added certificates for localhost where the CN and SAN match and differ.
2. Change subject name matching so the CN is checked if the SAN list doesn't exit, and only check the SAN list if present.

### CODE REVIEW

File: certs/test/gen-testcerts.sh [hunk 1]
```diff
@@ -95,3 +95,9 @@ generate_test_cert server-badaltnull www.nomatch.com DER:30:0d:82:0b:6c:6f:63:61
 
 # Generate Bad Alt Name CN=www.nomatch.com, Alt=www.nomatch.com
 generate_test_cert server-badaltname www.nomatch.com www.nomatch.com
+
+# Generate Good Alt Name CN=localhost, Alt=localhost
+generate_test_cert server-localhost localhost localhost
```
Comment (dgarske) [inline certs/test/gen-testcerts.sh]: Should/can you add a success test case for the new "server-localhost" cert?

**Comment by dgarske** [comment]
@toddouska : Would you mind reviewing this PR as well?

File: src/internal.c [hunk 2]
```diff
@@ -9170,6 +9193,7 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                             ret = DOMAIN_NAME_MISMATCH;
```
Comment (toddouska) [inline src/internal.c]: Let's put a MSG here about common name failure, otherwise looks good.

Comment (ejohnstown) [reply src/internal.c]: If the domain name isn't the common name, line 9164 (left) will log the common name failure, and if the domain name isn't in the subject alt names, you get an additional message on 9168 (left).

Comment (toddouska) [reply src/internal.c]: Ok, thanks, wasn't showing up on the right side.

**Pull request merged**